common/+deviceDrivers/YokoGS200.m

```diff
@@ -68,13 +68,13 @@
             %rate in V/s
             CurrentV = str2double(obj.query(':SOURCE:LEVEL?'));
             DeltaV = Vset-CurrentV;
-            step = rate*1e3; %1 ms step
+            step = rate*1e-3; %1 ms step
+            nsteps = round(abs(DeltaV/step));
             for j=1:nsteps
                 CurrentV=CurrentV+j*step;
                 obj.write([':SOURCE:LEVEL ' num2str(CurrentV)]);
-                pause(1e-3)
+                pause(0.6e-3) %each set instruction takes ~0.4 ms
             end
-            obj.write([':SOURCE:LEVEL ' num2str(Vset)]);
```

Comment:
Why did you remove this? It seems you might not end up at Vset. Say `CurrentV = 0` is 0 and `Vset = 3`  then `DeltaV=3` and if `step = 2` then you'll get two steps and end up at 4.